test/e2e/receive_test.go

```diff
@@ -12,97 +13,123 @@ import (
+		// receiver in the hashring than the one handling the request.
+		// The querier queries all the receivers and the test verifies
+		// the time series are forwarded to the correct receive node.
+
+		receiveHTTP1, receiveHTTP2, receiveHTTP3 := a.New(), a.New(), a.New()
+
+		r1 := receiver(receiveHTTP1, a.New(), a.New(), 1, remoteWriteEndpoint(receiveHTTP1), remoteWriteEndpoint(receiveHTTP2), remoteWriteEndpoint(receiveHTTP3))
+		r2 := receiver(receiveHTTP2, a.New(), a.New(), 1, remoteWriteEndpoint(receiveHTTP1), remoteWriteEndpoint(receiveHTTP2), remoteWriteEndpoint(receiveHTTP3))
+		r3 := receiver(receiveHTTP3, a.New(), a.New(), 1, remoteWriteEndpoint(receiveHTTP1), remoteWriteEndpoint(receiveHTTP2), remoteWriteEndpoint(receiveHTTP3))
+
+		prom1 := prometheus(a.New(), defaultPromRemoteWriteConfig(nodeExporterHTTP(1), remoteWriteEndpoint(r1.HTTP)))
+		prom2 := prometheus(a.New(), defaultPromRemoteWriteConfig(nodeExporterHTTP(2), remoteWriteEndpoint(r2.HTTP)))
+		prom3 := prometheus(a.New(), defaultPromRemoteWriteConfig(nodeExporterHTTP(3), remoteWriteEndpoint(r3.HTTP)))
+
+		q1 := querier(a.New(), a.New(), []address{r1.GRPC, r2.GRPC, r3.GRPC}, nil)
+
+		testCases = append(testCases, receiveTestConfig{
+			name:         "hashring",
+			cmds:         []scheduler{q1, prom1, prom2, prom3, r1, r2, r3},
+			queryAddress: q1.HTTP,
+			metrics: []model.Metric{
+				{
+					"__name__": "up",
+					"instance": model.LabelValue(nodeExporterHTTP(1)),
+					"job":      "node",
+					"receive":  "true",
+					"replica":  model.LabelValue("2"),
+				},
+				{
+					"__name__": "up",
+					"instance": model.LabelValue(nodeExporterHTTP(2)),
+					"job":      "node",
+					"receive":  "true",
+					"replica":  model.LabelValue("3"),
+				},
+				{
+					"__name__": "up",
+					"instance": model.LabelValue(nodeExporterHTTP(3)),
+					"job":      "node",
+					"receive":  "true",
+					"replica":  model.LabelValue("1"),
+				},
+			}})
+	}
+	{
+		// The replication suite creates three receivers but only one
+		// Prometheus that remote-writes data. The querier queries all
+		// receivers and the test verifies that the time series are
+		// replicated to all of the nodes.
+		receiveHTTP1, receiveHTTP2, receiveHTTP3 := a.New(), a.New(), a.New()
+
+		r1 := receiver(receiveHTTP1, a.New(), a.New(), 3, remoteWriteEndpoint(receiveHTTP1), remoteWriteEndpoint(receiveHTTP2), remoteWriteEndpoint(receiveHTTP3))
+		r2 := receiver(receiveHTTP2, a.New(), a.New(), 3, remoteWriteEndpoint(receiveHTTP1), remoteWriteEndpoint(receiveHTTP2), remoteWriteEndpoint(receiveHTTP3))
+		r3 := receiver(receiveHTTP3, a.New(), a.New(), 3, remoteWriteEndpoint(receiveHTTP1), remoteWriteEndpoint(receiveHTTP2), remoteWriteEndpoint(receiveHTTP3))
+
+		prom1 := prometheus(a.New(), defaultPromRemoteWriteConfig(nodeExporterHTTP(1), remoteWriteEndpoint(r1.HTTP)))
+		prom2 := prometheus(a.New(), defaultPromConfig("no-remote-write", 2, prom1.HTTP))
```

Comment:
previously, every receive component also created a prometheus, that is why we had two proms with `no-remote-write`. Now that they are independent, we can eliminate `prom2` and `prom3` from here. They are not doing anything

Reply:
:man_facepalming: